control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -388,7 +395,7 @@ type KubeAPIServerConfigParams struct {
 	InternalRegistryHostName     string
 	ExternalRegistryHostNames    []string
 	DefaultNodeSelector          string
-	AdvertiseAddress             string
+	AdvertiseAddress             []string
```

Comment:
AdvertiseAddresses?

Reply:
Nope, I've mixed both AdvertiseAddresses and AdvertiseAddress in the same param. The behaviour should remain the same because for now we are setting only 1 ip address. We can change the logic to support multi address in this parameter when KAS has support for that (https://github.com/kubernetes/enhancements/issues/2438)

Reply:
even though it holds only 1 ip, I think it should be renamed to AdvertiseAddresses since we changed the type to a slice.

Reply:
Ok makes sense. I've modified the code and tests accordingly